pandas/core/groupby/generic.py

```diff
@@ -701,111 +701,112 @@ def value_counts(
 
         ids, _, _ = self.grouper.group_info
         val = self.obj._values
+        codes = self.grouper.reconstructed_codes  # this will track the groups
 
         # groupby removes null keys from groupings
         mask = ids != -1
         ids, val = ids[mask], val[mask]
+        if dropna:
+            mask = ~isna(val)
+            if not mask.all():
+                ids, val = ids[mask], val[mask]
```

Comment:
codecov reports no testing here

Reply:
Most features are tested in test_series_groupby_value_counts, which is parameterized and includes dropna as a parameter.  That should address the below case as well (since it includes seed_nans in the dataframe generation.  I could add specific tests for these cases, but it seems like they are covered in the existing tests.

Reply:
@simonjayhawkins IIUC codecov gets results from the travis-37-cov build, which runs with `not slow`, so we wouldn't expect GroupBy.value_counts to show up there